server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
I think that if the node crashes after persisting these segments, it will include these persisted hydrants when starting back up, but will use kafka offsets from earlier segments. So the data will be ingested twice.